src/appshell/qml/Preferences/ScrollablePreferencesPage.qml

```diff
@@ -0,0 +1,112 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2021 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+
+import MuseScore.Ui 1.0
+import MuseScore.UiComponents 1.0
+
+PreferencesPage {
```

Comment:
It would be great to have only PreferencesPage without ScrollablePreferencesPage. It looks like PreferencePage can decide whether it can be scrolled or not (and show/hide gradients after that)

Reply:
In this context, "Scrollable" does not pre se mean that the content height is taller than the available. If that was the case, it would indeed be obvious that one PreferencesPage type is enough.

On "Scrollable" pages, the content height is determined by the contents. In practice, the page may or may not be scrollable, depending on whether the content height is taller than the available height. This applies to most pages, including the "General" and "Appearance" pages.

On "Non-Scrollable" pages, the content height is determined by the available height. So, the view will never be scrollable but will be stretched to fill the available height. This applies to the "MIDI device mappings" page, the "Shortcuts" page and the "Advanced" page.

Of course, it is _our choice_ whether the contents of a page should take its intrinsic size, or that it fills up the available height. So this can't be determined by the PreferencesPage.

However, I simplified it now, so that there is only one PreferencesPage component, with a property to control the behaviour. This seems like a more elegant solution.

Reply:
If we decide to add more items to the page, then we will need to control scrolling with this flag. Previously, this was done automatically(by contentHeight: content.height).

So, I agree with Roman
> It looks like PreferencePage can decide whether it can be scrolled or not (and show/hide gradients after that)

Reply:
Again, that is not the point of the `contentDeterminesContentHeight` flag. I think that we are agreeing all three, but that there is a misunderstanding.

If you add more content to a normal page, it will still automatically become scrollable if the size of the window is too small. So, in fact, nothing changes regarding that. Before, we would have to write `contentHeight: content.height`, but that is not even necessary anymore: it goes automatically.

But on some pages, we do _not_ want that the content height is determined based on the implicit height of the content, but we want that the content takes all available height, not more, not less. For example, the MIDI Device mappings page. Before, we would write `contentHeight: height` on those pages (where `height` is the height of the Flickable). Now, we can write `contentDeterminesContentHeight: false` instead, which is technically more characters, but its meaning is clearer.

So, in code, the idea is the following: 

```diff
 // GeneralPreferencesPage.qml and many others
 // content height determined by content itself
 PreferencesPage {
-    contentHeight: content.height
     
     Column {
         id: content
         ...
     }
 }
 
 // MidiDeviceMappingsPreferencesPage.qml, ShortcutsPreferencesPage.qml, AdvancedPreferencesPage.qml
 // content height follows available height
 PreferencesPage {
-    contentHeight: height // `height` is the height of the preferences page
+    contentDeterminesContentHeight: false
     
     PageContent {
         anchors.fill: parent
     }
 }
```

I hope this explanation is more understandable...

Maybe it would be clearer if I replace the `contentDeterminesContentHeight` property with a property called `contentFillsAvailableHeight`. So instead of `contentDeterminesContentHeight: false`, we would write `contentFillsAvailableHeight: true` on those exception pages.

Reply:
Indeed, I changed it to `contentFillsAvailableHeight: true`. I hope it is clear now.